### PR TITLE
Supports building and testing on Apple Silicon

### PR DESCRIPTION
**Pull request by brennanMKE**
*Issue #, if available:*

#1547 

A special testing dependency is able catch fatal errors but it requires changes to work with Apple Silicon. It has been updated to support the new M1 chips and 2 new test functions have been added to AmplifyTestCommon.

* `XCTAssertThrowFatalError`
* `XCTAssertNoThrowFatalError`

These functions use CwlPreconditionTesting and provide an API which matches XCTest so that it is familiar. All direct imports of CwlPreconditionTesting have been removed so it is only used from AmplifyTestCommon to support these 2 functions. Apple's API will show in a different color in Xcode than custom functions which are added in this code base.

## Flaky Tests

There were several tests which were failing sporadically and each were using `DispatchSemaphore` instead of an `XCTestExpectation` and so these were changed out and now the tests run reliably.

*Check points: (check or cross out if not relevant)*

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [x] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 ## Unreleased
+
+### Build Fixes
+
+- **Apple Silicon**: Adds support to build and run tests on M1 Macs
+
```
Comment (diegocstn) [inline CHANGELOG.md:7]: this might not be necessary as the changelog is automatically updated by the release process

**Review by diegocstn — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (brennanMKE) [reply CHANGELOG.md:7]: I could take it out.

File: AmplifyTestCommon/Helpers/AmplifyAssertions.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+
+@testable import CwlPreconditionTesting
+
+public func XCTAssertThrowFatalError(_ expression: @escaping () -> Void,
```
Comment (royjit) [inline AmplifyTestCommon/Helpers/AmplifyAssertions.swift:13]: Is this different from https://developer.apple.com/documentation/xctest/1500795-xctassertthrowserror

Comment (brennanMKE) [reply AmplifyTestCommon/Helpers/AmplifyAssertions.swift:13]: Yes, it is meant to look like those XCTest functions. This one catches fatal errors like the CwlPreconditionTesting package does.

**Pull request merged**